hail/src/main/scala/is/hail/expr/ir/functions/GenotypeFunctions.scala

```diff
@@ -10,32 +10,32 @@ import is.hail.expr.types.virtual.{TArray, TFloat64, TInt32, Type}
 object GenotypeFunctions extends RegistryFunctions {
 
   def registerAll() {
-    registerCode("gqFromPL", TArray(tv("N", "int32")), TInt32, (_: Type, _: PType) => PInt32()) { case (r, rt, (tPL: PArray, _pl: Code[Long])) =>
-      val pl = r.mb.newLocal[Long]("pl")
-      val m = r.mb.newLocal[Int]("m")
-      val m2 = r.mb.newLocal[Int]("m2")
-      val len = r.mb.newLocal[Int]("len")
-      val pli = r.mb.newLocal[Int]("pli")
-      val i = r.mb.newLocal[Int]("i")
-      Code(
-        pl := _pl,
-        m := 99,
-        m2 := 99,
-        len := tPL.loadLength(pl),
-        i := 0,
-        Code.whileLoop(i < len,
-          tPL.isElementDefined(pl, i).mux(
-            Code._empty,
-            Code._fatal[Unit]("PL cannot have missing elements.")),
-          pli := Region.loadInt(tPL.loadElement(pl, len, i)),
-          (pli < m).mux(
-            Code(m2 := m, m := pli),
-            (pli < m2).mux(
-              m2 := pli,
-              Code._empty)),
-          i := i + 1
-        ),
-        m2 - m)
+    registerPCode("gqFromPL", TArray(tv("N", "int32")), TInt32, (_: Type, _: PType) => PInt32())
+    { case (r, rt, _pl: PIndexableCode) =>
+      val code = EmitCodeBuilder.scopedCode(r.mb) { cb =>
+        val pl = _pl.memoize(cb, "plv")
+        val m = cb.memoize[Int](const(99), "m")
```

Comment:
not specific to this PR, but why is this called memoize?

Reply:
I see from the perspective that we don't just re-run the first argument, but store it instead. That's a bit different than what I think of as memoization (checking if a value is already stored, say in a hash table by looking up some key), our memoization is just an assignment.